src/sqlfluff/core/linter/linter.py

```diff
@@ -542,8 +543,27 @@ def lint_fix_parsed(
                     "loops."
                 )
                 break
-        if fix and loop + 1 == loop_limit:
-            linter_logger.warning(f"Loop limit on fixes reached [{loop_limit}].")
+        else:
+            if fix:
```

Comment:
The old `if` is considered a poor, hacky way to determine if the loop exited or not. I updated it to use `for/else` instead. Explanation: https://book.pythontips.com/en/latest/for_-_else.html#else-clause